src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -47,33 +73,55 @@ export const MultiSelect: FC<MultiSelectProps> = ({
     ariaLabel = 'Select list',
     disabled = false,
     placeholder,
+    filterLabel,
+    noResultsLabel,
     type = MultiSelectType.Default,
     size = MultiSelectSize.Medium,
     validation = Validation.Default,
+    emphasis = MultiSelectEmphasis.Default,
+    decorator = null,
+    filterable = false,
 }) => {
     const [open, setOpen] = useState(false);
+    const [checkboxes, setCheckboxes] = useState<Item[]>(items.map((item) => ({ ...item, label: item.value })));
+    const hasResults = !!checkboxes.find((item) => !item.isCategory && !item.isDivider);
     const overlayRef = useRef<HTMLDivElement | null>(null);
     const triggerRef = useRef<HTMLDivElement | null>(null);
+    const multiSelectRef = useRef<HTMLDivElement | null>(null);
+    const menuRef = useRef<HTMLDivElement | null>(null);
+    const filterInputRef = useRef<HTMLInputElement | null>(null);
     const { isFocusVisible, focusProps } = useFocusRing();
-    const checkboxes = items.map((item) => ({ ...item, label: item.value }));
+
+    const hasSelectedItems = activeItemKeys.length > 0;
+    const summarizedLabel = [activeItemKeys.length, 'selected'].join(' ');
+    const inputWidth = getInputWidth(hasSelectedItems, filterLabel, placeholder);
+
+    const handleClickOutside = () => {
+        if (emphasis === MultiSelectEmphasis.Weak) {
+            setOpen(false);
+        }
+    };
+
+    useClickOutside(menuRef?.current, handleClickOutside, [multiSelectRef?.current as HTMLElement]);
+
+    const toggleOpen = () => setOpen((open) => !open);
```

Comment:
The `ESC` key when focusing on the menu was supposed to close the menu but it is not working here. You can see it working on the `Menu` component stories